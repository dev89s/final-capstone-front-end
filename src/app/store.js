import { configureStore } from '@reduxjs/toolkit';

import dataReducer from '../components/Home/createSlice';
import userReducer from './user/userSlice';
import roomReducer from './room/roomSlice';

const store = configureStore({
  reducer: {
    user: userReducer,
    dataSlice: dataReducer,
    room: roomReducer,
  },
});

export default store;
