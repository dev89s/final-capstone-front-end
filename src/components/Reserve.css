.reserve-container {
  position: relative;
  background-image: url('/public/pexels-iv√°n-rivero-1001965.jpg'); /* Replace with your image URL */
  background-size: cover;
  background-position: center;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  overflow: hidden; /* Add this line */
}

.background-overlay {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(44, 234, 44, 0.5); /* Green color with transparency */
}

.reservation-form-container {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(255, 255, 255, 0.9);
  border: 0 solid #ccc;
  padding: 0;
  border-radius: 20px;
  box-shadow: 8px 0 5px rgba(0, 0, 0, 0.2);
  width: 50vw;
  z-index: 1;
}

.input-row {
  display: flex;
  justify-content: space-between;
}

.error {
  color: red;
  font-size: 0.875rem;
  margin-top: 5px;
}

.reservation-form {
  width: 90%;
  padding: 15px;
}

.reservation-form h1 {
  font-size: 1.5rem;
  margin-bottom: 20px;
  text-align: center;
}

.reservation-form input,
select {
  width: 100%;
  padding: 7px;
  margin-bottom: 15px;
  border: 1px solid #ccc;
  border-radius: 43px;
  font-size: 1rem;
}

.reservation-form button {
  width: 100%;
  padding: 10px;
  background-color: #37eb46;
  color: white;
  border: none;
  border-radius: 4px;
  font-size: 1rem;
  cursor: pointer;
}

.reservation-form button:hover {
  background-color: #20ae36;
}

@media (max-width: 900px) {
  .input-row {
    flex-direction: column;
  }
}
