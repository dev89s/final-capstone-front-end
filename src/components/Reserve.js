/* eslint-disable camelcase */
import axios from 'axios';
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import './Reserve.css';

const Reserve = () => {
  const { cities } = useSelector((store) => store.cities);

  const [room_id, setRoomId] = useState('');
  const [date, setDate] = useState('');
  const [city_id, setCityId] = useState('');
  const [error, setError] = useState(false);

  const navigate = useNavigate();
  const user = useSelector((store) => store.user.userInfo);

  const handleReserve = (e) => {
    e.preventDefault();
    setError(false);
    if (room_id !== '' && date !== '' && city_id !== '') {
      axios
        .post('http://localhost:3000/api/v1/reservations', {
          reservation: {
            room_id,
            date,
            city_id,
            user_id: user.id,
          },
        })
        .then((response) => {
          if (response.status === 201) {
            setRoomId('');
            setDate('');
            setCityId('');
            navigate('/my_reservations');
          }
        })
        .catch((err) => err && setError(true));
    } else {
      setError(true);
    }
  };

  const renderCities = () => cities.map((city) => (
    <option key={city.id} value={city.id}>
      {city.name}
    </option>
  ));

  return (
    <div className="reserve-container">
      <div className="background-overlay"> </div>
      <div className="form-container">
        <form onSubmit={handleReserve} className="reservation-form">
          <h1>Reserve Room</h1>
          <div className="input-row">
            <div className="input-container-1">
              <input
                type="text"
                value={room_id}
                onChange={(e) => setRoomId(e.target.value)}
                placeholder="Room ID"
                required
              />
            </div>
            <div className="input-container-3">
              <input
                type="date"
                value={date}
                onChange={(e) => setDate(e.target.value)}
                placeholder="Date"
                required
              />
            </div>
            <div className="input-container-2">
              <select
                type="text"
                value={city_id}
                onChange={(e) => setCityId(e.target.value)}
                placeholder="City ID"
                required
              >
                {renderCities()}
              </select>
            </div>
          </div>

          {error && <span className="error">Please fix the error</span>}

          <button type="submit">Reserve Room</button>
        </form>
      </div>
    </div>
  );
};

export default Reserve;
