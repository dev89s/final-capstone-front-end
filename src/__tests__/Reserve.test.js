import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Reserve from '../components/Reserve';
import store from '../app/store';

describe('Should render the reserver page', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Reserve />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have the title text', () => {
    render(
      <Provider store={store}>
        <Reserve />
      </Provider>,
      { wrapper: BrowserRouter },
    );
    const text = screen.getByText('Select City');
    expect(text).toBeInTheDocument();
  });
});
