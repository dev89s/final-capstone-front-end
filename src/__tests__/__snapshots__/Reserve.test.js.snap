// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the reserver page should render correctly 1`] = `
<div
  className="reserve-container"
>
  <div
    className="background-overlay"
  >
     
  </div>
  <div
    className="reservation-form-container"
  >
    <form
      className="reservation-form"
      onSubmit={[Function]}
    >
      <h1>
        Reserve Room
      </h1>
      <div
        className="input-row"
      >
        <div
          className="input-container-1"
        >
          <select
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          >
            <option
              disabled={true}
              value=""
            >
              Select Room
            </option>
          </select>
        </div>
        <div
          className="input-container-3"
        >
          <input
            onChange={[Function]}
            placeholder="Date"
            required={true}
            type="date"
            value=""
          />
        </div>
        <div
          className="input-container-2"
        >
          <select
            onChange={[Function]}
            placeholder="City ID"
            required={true}
            type="text"
            value=""
          >
            <option
              disabled={true}
              value=""
            >
              Select City
            </option>
          </select>
        </div>
      </div>
      <button
        type="submit"
      >
        Reserve Room
      </button>
    </form>
  </div>
</div>
`;
